trino/sqlalchemy/compiler.py

```diff
@@ -147,6 +147,29 @@ def visit_BLOB(self, type_, **kw):
     def visit_DATETIME(self, type_, **kw):
         return self.visit_TIMESTAMP(type_, **kw)
 
+    def visit_TIMESTAMP(self, type_, **kw):
+        datatype = "TIMESTAMP"
+        precision = getattr(type_, "precision", None)
+        if precision not in range(0, 12):
```

Comment:
`range(0,12)` mean `[0, 12)`
Trino is support precision up to `12` (picoseconds), [according the docs](https://trino.io/docs/current/language/types.html#time-p).

Reply:
ohh.. thanks for catching it